src/metabase/query_processor/util/nest_query.clj

```diff
@@ -24,7 +24,7 @@
    add/normalize-clause
    (lib.util.match/match (walk/prewalk (fn [x]
                                          (if (map? x)
-                                           (dissoc x :source-query :source-metadata)
+                                           (dissoc x :source-query :source-metadata :temporal-unit)
```

Comment:
It's possible that we want a broader fix here.  If other options can produce sql that changes the type of the field, they would cause similar issues. 
 Perhaps we want a whitelist instead of blacklisting certain fields?

Reply:
There are a few more cases like this, though they're handled somewhat differently than temporal units.

But I wouldn't worry too much about implementation details; this whole setup will be changing substantially with the new refs.